cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/workflow/CustomActionExecutor.java

```diff
@@ -86,7 +86,7 @@ void execute() throws Exception {
       customActionContext.setState(new ProgramState(ProgramStatus.COMPLETED, null));
 
     } catch (Throwable t) {
-      customActionContext.setState(new ProgramState(ProgramStatus.FAILED, Throwables.getRootCause(t).getMessage()));
+      customActionContext.setState(new ProgramState(ProgramStatus.FAILED, t.getMessage()));
```

Comment:
It is problematic in either case. Usually the actual message is hidden deep inside the root cause. However, contextual information is available in intermediate wrapping exceptions. Since we are not recording the whole exception stacktrace, but instead just a String message. It is better to concatenate the message in some meaningful way across the exception chain (some messages along the chain could be `null`, especially the one coming from Java itself).

Reply:
Fixed